pkg/commands/patch/patch_modifier.go

```diff
@@ -13,6 +13,20 @@ var (
 	patchHeaderRegexp = regexp.MustCompile(`(?ms)(^diff.*?)^@@`)
 )
 
+type PatchOptions struct {
+	// Create a reverse patch
+	Reverse bool
+
+	// If true, we're building a patch that we are going to apply using
+	// "git apply --reverse". In that case the filtering of partial hunks needs
+	// to be done a little differently.
+	WillBeAppliedReverse bool
```

Comment:
I'd like this comment to be more explicit about the difference between creating a reverse patch (i.e. what the `Reverse` option is for) and creating a patch to be applied in reverse. In fact it's just occurred to me that I don't properly understand the difference haha

Reply:
I added two fixups to try and make it clearer; does that help?

Reply:
Also added another fixup to assert that `Reverse` and `WillBeAppliedReverse` are not used together.

Reply:
@jesseduffield Could you have another look if you have a minute? Is there anything missing for moving this forward?

(Let me know if you'd like me to squash the fixups; one of them conflicts, so it's not just a matter of pressing 'S'.)

Reply:
I've spent some time looking into this some more to make sense of the change and I'm thinking we should combine these two flags into a single enum. Let me know what you think about https://github.com/jesseduffield/lazygit/commit/db9b3cbe19ff7c40b5f218a1cb2c8c7f0cd4070b

Reply:
Yes, combining the flags into an enum makes a lot of sense to me. I had to add two small fixes to make it all work, see 061fec2d31d57f59ee.

The history of the branch is a bit of a mess now, let me see if I can clean it up at least a little bit. Also, the parameter of `PatchManager.RenderPatchForFile` is still called `willBeAppliedReverse`, I think this should just be called `reverse`.